public/app/plugins/panel/calendarV2/module.tsx

```diff
@@ -0,0 +1,31 @@
+import {
+  PanelPlugin,
+} from '@grafana/data';
+
+export const plugin = new PanelPlugin<CalendarOptions, CalendarConfig>(CalendarDasboard).setExplorePanel(CalendarExplore)
+
+function Calendar(props: CalendarProps) {
+  return "Calendar"
+}
+
+/**
+ * This one will be used in dashboard context
+ */
+function CalendarDashboard(props: PanelProps<CalendarOptions>) {
+  // use dashboard props to configure calendar component
+  return <Calendar/>
+}
+
+/**
+ * This one will be used in explore context.
+ *
+ * This alone isn't enough to be able to use this in explore. It still needs to have visualisationType set in plugin.json
+ * otherwise we will never match it to any data.
+ */
+function CalendarExplore(props: ExplorePanelProps) {
+  // use explore props to configure calendar component like some added interactivity
+  return <Calendar/>
+}
```

Comment:
Why can't these be the same, there is nothing Dashboard specific with PanelProps, it's designed to be agnostic and can be used anywhere (Used from new alerting view for example and in other places via PanelRenderer)

Reply:
If you need specific explore actions, it could be done with events or maybe PanelContext ? it has a app property so the panel can behave differently if shown in Explore: 

https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/PanelChrome/PanelContext.ts

Reply:
@torkelo hmm not sure what you mean there is nothing specific with PanelProps for dashboards. There are specific props that aren't available in dashboards but are available in Explore. Putting all of that in props or a context means that the type of that is messy. Like even in the context there already is onSplitOpen prop which is optional, but it's not optional in Explore. This creates a single type where lots of things are optional which isn't true as in reality it's 2 types with all of the props required (just 2 different sets). That is also why there is the `setExplorePanel`, it could be solved in different ways but this imho gives the most reasonable typings of all the options we discussed for this. cc @ryantxu as this sort of ties to your question.